### DISCUSSION TITLE
mmmm yes. more rust

### DISCUSSION THREAD

**Discussion by ThGrSoRu** [sxyazi/yazi#419] [4. Ideas]
I recently discovered another [awesome rust project](https://github.com/ouch-org/ouch).
This is something like unarchiver, supporting only popular formats, with rar and 7z being added recently.

I just wanted to give this some limelight, as I think this fits nicely with yazi.

**Comment by sxyazi** [comment]
I'm currently not very satisfied with unar, 7zip, or ouch. Maybe we need to implement our own, but thanks for your information.

**Reply by ThGrSoRu** [reply]
What do you feel is missing from all these?

**Reply by sxyazi** [reply]
Only `unar` provides JSON-formatted directory structure output, but it can't retrieve directories at a specific level; instead, it fetches all at once, which is wasteful of resources.

Moreover, parsing JSON itself is wasteful imo. Hence we need to create our own tool, maybe based on one of them, to efficiently fetch those data.
